lambdas/services/document_manifest_job_service.py

```diff
@@ -0,0 +1,165 @@
+import itertools
+import os
+
+from enums.lambda_error import LambdaError
+from enums.supported_document_types import SupportedDocumentTypes
+from enums.zip_trace import ZipTraceStatus
+from models.document_reference import DocumentReference
+from models.zip_trace import DocumentManifestJob, DocumentManifestZipTrace
+from pydantic import ValidationError
+from services.base.dynamo_service import DynamoDBService
+from services.base.s3_service import S3Service
+from services.document_service import DocumentService
+from utils.audit_logging_setup import LoggingService
+from utils.common_query_filters import UploadCompleted
+from utils.lambda_exceptions import DocumentManifestJobServiceException
+from utils.utilities import get_file_key_from_s3_url
+
+logger = LoggingService(__name__)
+
+
+class DocumentManifestJobService:
+    def __init__(self):
+        create_document_aws_role_arn = os.getenv("PRESIGNED_ASSUME_ROLE")
+        self.s3_service = S3Service(custom_aws_role=create_document_aws_role_arn)
+        self.document_service = DocumentService()
+        self.dynamo_service = DynamoDBService()
+        self.zip_output_bucket = os.environ["ZIPPED_STORE_BUCKET_NAME"]
+        self.zip_trace_table = os.environ["ZIPPED_STORE_DYNAMODB_NAME"]
+        self.documents: list[DocumentReference] = []
+
+    def create_document_manifest_job(
+        self,
+        nhs_number: str,
+        doc_types: list[SupportedDocumentTypes],
+        selected_document_references: list[str] = None,
+    ) -> str:
+        logger.info("Retrieving Document References for manifest")
+
+        self.documents = list(
+            itertools.chain.from_iterable(
+                [
+                    self.document_service.fetch_available_document_references_by_type(
+                        nhs_number=nhs_number,
+                        doc_type=doc_type,
+                        query_filter=UploadCompleted,
+                    )
+                    for doc_type in doc_types
+                ]
+            )
+        )
```

Comment:
Actually we've got a helper util function `flatten` for this purpose :)
`from utils.utilities import get_file_key_from_s3_url, flatten`

```suggestion
        self.documents = flatten(
            [
                self.document_service.fetch_available_document_references_by_type(
                    nhs_number=nhs_number,
                    doc_type=doc_type,
                    query_filter=UploadCompleted,
                )
                for doc_type in doc_types
            ]
        )
```